src/bun.js/bindings/root.h

```diff
@@ -102,4 +102,10 @@
 #define ZIG_EXPORT(...)
 #define ZIG_NONNULL
 
+#if USE(MIMALLOC)
+#define BUN_FREE(ptr) mi_free(ptr)
+#else
+#define BUN_FREE(ptr) free(ptr)
+#endif
```

Comment:
I checked the type signature of `mi_free`:

```cpp
void mi_free(void* p);
```

Would urge you to do something like

```cpp
inline void bunFree(void* ptr) {
    mi_free(ptr);
}
```